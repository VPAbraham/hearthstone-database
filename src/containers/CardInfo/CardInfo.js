import React from 'react';
import './CardInfo.scss';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const CardInfo = (props) => {
  const { selCard } = props;
  console.log(selCard)
  return (
    <section className='card-info'>
      
      <h1>Card Data for {selCard.name}</h1>
      <div className='info-container'>
        <img src={selCard.imageUrl} alt='card image'/>
        <div>
          <p>Pokedex No. {selCard.pokedexNum}</p>
          <p>HP: {selCard.hp}</p>
          <p>Rarity: {selCard.rarity}</p>
          <p>Set: {selCard.set}</p>
          <p>Art by: {selCard.artist}</p>
          <p>Type: {selCard.types[0]}</p>
          <p>Attack: {selCard.attacks[0].name}</p>

      

        </div>
      </div>
      <Link to='/'>
        <button>Return to Main Page</button>
      </Link>
    </section>
  )
}

export const mapStateToProps = ({ selCard }) => ({
  selCard
})

export default connect(mapStateToProps)(CardInfo);

CardInfo.propTypes = {
  selCard: PropTypes.object.isRequired 
}

