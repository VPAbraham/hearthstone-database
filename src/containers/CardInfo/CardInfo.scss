.card-info {
  font-family: 'Work Sans', sans-serif;
  text-align: center;

  h1 {
    text-align: center;
    font-size: 2em;
    height: 5rem;
    padding: 1rem;
    margin-bottom: 5vh;
  }

  .info-container {
    display: flex;
    justify-content: space-evenly;

    img {
      height: 20rem;
    }

    div {
      font-size: 1.2em;
      display: flex;
      flex-direction: column;
      justify-content: space-around;
      border: solid 1px rgb(163, 163, 163);
      border-radius: 5px;
      padding: 5px;
      background: rgb(26, 25, 25);
    }
  }

  button {
    height: 4rem;
    border-radius: 5px;
    width: 24vw;
    color: white;
    border: none;
    font-family: 'Montserrat', sans-serif;
    font-size: 1.8em;
    background: rgb(185, 88, 88);
    transition: 0.5s;
    margin-top: 4rem;
    margin-bottom: 5rem;

    &:hover {
      background: rgb(206, 159, 159);
      transition: 0.5s;
      box-shadow: 0px 5px 8px -4px #d48854;

    }
  }
}