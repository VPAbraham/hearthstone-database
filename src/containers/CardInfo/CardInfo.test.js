import React from 'react';
import { shallow } from 'enzyme';
import { CardInfo, mapStateToProps } from './CardInfo';

describe('CardInfo', () => {
  let wrapper;
  beforeEach(() => {
    const mockSelCard = {name: 'Cubone', type: 'Fighting', hp:'70'}

    wrapper = shallow(
      <CardInfo 
      selCard={mockSelCard}
      />
      )
  })
  it('should match the snapchot with data passed through', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return a selected card object', () => {
    const mockState = { selCard: { name: 'Cubone', type: 'Fighting', hp: '70' }};
    const expected = { selCard: { name: 'Cubone', type: 'Fighting', hp: '70' } };

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected);

  })
})