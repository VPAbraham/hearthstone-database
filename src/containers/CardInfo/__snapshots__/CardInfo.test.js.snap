// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardInfo should match the snapshot with data passed through 1`] = `
<section
  className="card-info"
>
  <h1>
    Card Data for 
    Cubone
  </h1>
  <div
    className="info-container"
  >
    <img
      alt="card image"
    />
    <div>
      <p>
        Pokedex No. 
      </p>
      <p>
        HP: 
        70
      </p>
      <p>
        Rarity: 
      </p>
      <p>
        Set: 
      </p>
      <p>
        Art by: 
      </p>
      <p>
        Type: 
        Fire
      </p>
      <p>
        Attack: 
      </p>
    </div>
  </div>
  <Link
    to="/"
  >
    <button>
      Return to Main Page
    </button>
  </Link>
</section>
`;
